Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandCharacterAutofixStuckInLocation.java

```diff
@@ -0,0 +1,70 @@
+package com.universeprojects.miniup.server.commands;
+
+import java.util.Map;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.universeprojects.miniup.CommonChecks;
+import com.universeprojects.miniup.server.GameUtils;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.WebUtils;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
+import com.universeprojects.miniup.server.commands.framework.Command.JavascriptResponse;
+
+/**
+ * Checks to see if a character is stuck in a location with no discoverable paths available. If the character
+ * is stuck, their locationKey is set to the parentLocationKey of their current location. 
+ * 
+ * @author papamarsh
+ * 
+ */
+public class CommandCharacterAutofixStuckInLocation extends Command {
+
+	public CommandCharacterAutofixStuckInLocation(ODPDBAccess db, HttpServletRequest request,
+			HttpServletResponse response) {
+		super(db, request, response);
+	}
+
+	@Override
+	public void run(Map<String, String> parameters) throws UserErrorMessage {
+		ODPDBAccess db = ODPDBAccess.getInstance(request);
+		CachedDatastoreService ds = getDS();
+		
+		//grab the current character, its location, and the location's parent
+		CachedEntity character = db.getCurrentCharacter();
+		CachedEntity currentLocation = db.getEntity((Key) character.getProperty("locationKey"));
+		CachedEntity parentLocation = db.getParentLocation(ds, currentLocation);
+		
+		//special case for if user is in hell
+		if(currentLocation==db.getLocationByName("Ninth Circle of Hell"))
+		{
+			throw new UserErrorMessage("Nice Try! But there's no escape from Hell :)");
+		}
+		
+		//list the paths connected to the character's location
+		List<CachedEntity> pathList = db.getPathsByLocation(currentLocation.getKey());
+		for(CachedEntity checkPath:pathList)
+		{
+			CachedEntity checkPathLocation1 = db.getEntity((Key) checkPath.getProperty("location1Key"));
+			CachedEntity checkPathLocation2 = db.getEntity((Key) checkPath.getProperty("location1Key"));

```

Comment:
Since this entire command is supposed to fail gracefully, maybe we should assume the locations could be null as well and skip a path if it's malformed..... actually, the easiest thing could be to wrap everything inside the for loop in a try-catch and just ignore any exceptions that do come up.